lib/srt/line.rb

```diff
@@ -41,7 +41,9 @@ def webvtt_time_str
     end
 
     def to_s(time_str_function=:time_str)
-      [sequence, (display_coordinates ? send(time_str_function) + display_coordinates : send(time_str_function)), text, ''].flatten.join("\n")
+      # ensure a new line is added for an empty text block
+      txt = text.empty? ? [""] : text
+      [sequence, (display_coordinates ? send(time_str_function) + display_coordinates : send(time_str_function)), txt , ''].flatten.join("\n")
```

Comment:
could we break this out a little bit more to have it be more readable overall? ex:

```
def to_s(time_str_function=:time_str)
  content = text.empty? ? [''] : text
  time_and_coordinates = (display_coordinates ? send(time_str_function) + display_coordinates : send(time_str_function))
  [sequence, time_and_coordinates, content, ''].flatten.join("\n")
end
```